src/components/Shared/ExpensesCostCenters.js

```diff
@@ -0,0 +1,172 @@
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import CustomTextField from '../Inputs/CustomTextField'
+import { GeneralLedgerRepository } from 'src/repositories/GeneralLedgerRepository'
+
+export default function ExpensesCostCenters({ labels, maxAccess, row, window, updateRow, recordId, readOnly }) {
+    const { formik } = useForm({
+        maxAccess,
+        enableReinitialize: true,
+        validateOnChange: true,
+        initialValues: {
+          costCenters: row.costCenters || [{
+            id: 1,
+            pvId: recordId || 0,
+            seqNo: 1,
+            ccSeqNo: '',
+            ccId: '',
+            amount: '',
+          }],
+          row,
+          balance: null,
+          amountAssigned: null,
+        },
+        validate: values => {
+            const errors = {}
+            if (values.balance != 0) {
+              errors.balance = ' '
+            }
+
+            return errors
+        },         
+        onSubmit: async obj => {
+            try {
+                const costCenters = obj.costCenters.map((costCenter, index) => ({
+                        ...costCenter,
+                        id: row.id,
+                        ccSeqNo: index + 1,
+                        seqNo: row.id,
+                        pvId: recordId
+                    })
+                );
+
+                updateRow({ changes: { costCenters } })
+                window.close()
+            } catch (error) { }
+        }
+    })
+            
+  const columns = [
+    {
+        component: 'resourcelookup',
+        label: labels.costCenter,
+        name: 'reference',
+        props: {
+          valueField: 'reference',
+          displayField: 'reference',
+          displayFieldWidth: 2,
+          endpointId: GeneralLedgerRepository.CostCenter.snapshot,
+          mapping: [
+            { from: 'recordId', to: 'ccId' },
+            { from: 'name', to: 'ccName' },
+            { from: 'reference', to: 'ccRef' }
+          ],
+          columnsInDropDown: [
+            { key: 'reference', value: 'Reference' },
+            { key: 'name', value: 'Name' }
+          ],
+          readOnly
+        }
+      },
+      {
+        component: 'textfield',
+        label: labels.name,
+        name: 'ccName',
+        props: {
+          readOnly: true
+        }
+      },
+      {
+        component: 'textfield',
+        label: labels.amount,
+        name: 'amount',
+        props: {
+          readOnly
+        }
+      },
+  ]
+
+
+    const totalAmount = formik.values?.costCenters?.reduce((amount, row) => {
+      const amountValue = parseFloat(row.amount?.toString().replace(/,/g, '')) || 0
+
+      return amount + amountValue
+    }, 0)
+
+
+    const balance = row.amount - totalAmount
+
+    useEffect(() => {
+        formik.setFieldValue('balance', balance)
```

Comment:
Is balance a changeable value? or readonly? If readonly (and we don't submit it to the api), not need to place it inside formik.